### PR TITLE
Address Overhaul Feedback

### PR DESCRIPTION
**Pull request by iCollin**
Address feedback on the sdl_core_guides overhaul project.

Changelog (unchecked means to-do)

# Core Guides

- [x] The first page header should match the title in the navigation section
- [x] Ensure each section naming templates are similar. Some sections have "Guide" appended some don't.
- [ ] One thing that is very noticeable is that each guide is written by someone else. I think after this release we could spend some time focused on creating a more "common voice" between sections.

#### Installation
- [x] Take out the mention of creating a build folder. This is discussed later and at this point it is confusing without further explanation. 
- [x] Add the git command in a code block for cloning the repo

#### Dependencies

- [x] Add text that states more options for the CMakeLists.txt file will be discussed later in the guide.
- but are more options discussed later?

#### Building

- [x] change the following
First, create a build folder separate from your source folder, for example: 
to
Begin by creating a build folder outside of SDL Core source folder, for example:

#### Start SDL Core

- [x] change the following
Once SDL Core is compiled and installed you can start it from the executable in the bin folder
to
Once SDL Core is compiled and installed you can start it from the executable in the newly created bin folder under your build folder directory

### INI Configuration

- [x] Change the first header from smartDeviceLink.ini to INI Configuration
- [x] Add a section about how to change these options and that Core will need to be restarted to have them take effect. 
- [x] Maybe add a note that as the guides progress, some of these options will be discussed in greater detail.

### Multiple Transport Configuration

- [x] The first paragraph has "Multiple Transports` but the T doesn't need to be capitalized
- both M&T are capitalized throughout the guide
- [x] A link to the larger guide should be added. 
- a link exists to the feature documentation guide

### SDL Core and HMI Communication

- [x] In the "Must" section it references "SDL" as an object, but it should be "SDL Core"

##### Request

- [x] The Key values in the table don't need quotes
- [x] The "componentNames" section could use more flushing out. Describing in greater detail what each component is responsible for or linking to sections that do just that.
- [x] Add verbiage that describes that the UI is simply one component of the HMI and that the other components of the HMI can stand alone and outside the UI HMI component.

### Creating the HMI UI

- [x] Creating the HMI UI "Component" ?
- [x] There is a lot of information in this section. I would really like to see some visuals. This would include a sequence diagram that builds over the course of the section.
- [x] It would also be good to see visuals of the actual UI beyond just the app list.

#### First Steps

- [x] Remove this First Steps header, it's redundant

#### Supported Template Views
- [x] Possibly link out to the RPC spec for the `PredefinedLayouts`? Or the RPC Spec in the overview guides which should prevent broken links.

##### TILES MENU EXAMPLE
- [ ] Should this have images in the tiles? It looks odd with just text positioned at the bottom of the square.

### Encryption Setup

- [x] Might be a good idea to add a little intro into encryption. The guide just starts right up with how to set it up without ever explaining or linking to other sections that describe how it works or what it's for.
- [x] Also, adding a link to the policy table sections would be a good idea to bring the concept full circle.
- [x] Is OpenSSL already part of the dependencies or will the developer have to install that?
- [ ] I think adding a new section to the Overview guides that describes encryption will be a good idea. I don't think it's really described in detail anywhere other than the protocol spec itself and that's tough to read through.

## Transport Manager Programming

- [x] Do we need the note about UML refresher under both diagrams? Probably not a big deal to keep.
- I think it's good for this to be easy to find, it does take up a lot of page space though

### Audio and Video Streaming Setup
- [x] Might be a good idea to add a note that the generic_hmi project does not currently support video streaming.
- [x] This guide could use some work making it flow better. There's a lot of individual sections, but I think it needs to start with a better description of what steps to take in order and what you can do in place of another.

#### HMI Modifications
- [x] Can we add language that this will be discussed further in the guide? The single sentence seems a little out of place. 
- Section was moved to the top of the guide

##### GSTREAMER Setup

- [x] Why is Ubuntu 14.04+ the reference here? Shouldn't we be consistent? 

### App Service Provider Guide

- [x] Should there be an App Service Consumer guide as well? Might really help OEMs understand how to integrate.
- [x] Add a link to the overview guides to describe App Services in greater detail as a feature.

### CODE REVIEW

File: docs/Feature Documentation/Audio and Video Streaming Guide/index.md [hunk 1]
```diff
@@ -31,54 +21,46 @@ Comment out the following lines in `app/model/sdl/Abstract/Model.js`:
...
+If you're streaming video over TCP, you can point gstreamer directly to your phone's stream using
 ```
 gst-launch-1.0 tcpclientsrc host=<Device IP Address> port=3000 ! decodebin ! videoconvert ! ximagesink sync=false
 ```
 
 ## Pipe Streaming
 
 ### Configuration (smartDeviceLink.ini)
-In the Core build folder, open `bin/smartDeviceLink.ini` and the make the following changes:
+In the Core build folder, open `bin/smartDeviceLink.ini` and ensure the following values are set:
```
Comment (jacobkeeler) [inline docs/Feature Documentation/Audio and Video Streaming Guide/index.md:43]: This same sentence needs to be changed in several places in the guide

File: docs/Getting Started/INI Configuration/index.md [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
 
 ## Sections
 
+!!! NOTE
+As the guides progress, some of these sections will be discussed in greater detail.
+!!!
+
 * `HMI` - Settings relating to the HMI connection, including server and port information.
-* `MEDIA MANAGER` - Settings related to media features (audio/video streaming and audio pass thru). Several of these options are described in more detail in the [Audio/Video Streaming Setup Guide](../../feature-documentation/audio-and-video-streaming-setup/).
+* `MEDIA MANAGER` - Settings related to media features (audio/video streaming and audio pass thru). Several of these options are described in more detail in the [Audio/Video Streaming Guide](../../feature-documentation/audio-and-video-streaming-guide/).
```
Comment (jacobkeeler) [inline docs/Getting Started/INI Configuration/index.md:35]:
```suggestion
* `MEDIA MANAGER` - Settings related to media features (audio/video streaming and audio pass thru). Several of these options are described in more detail in the [Audio/Video Streaming Guide](../../feature-documentation/audio-and-video-streaming/).
```

File: docs/Feature Documentation/Audio and Video Streaming Guide/index.md [hunk 3]
```diff
@@ -31,54 +21,46 @@ Comment out the following lines in `app/model/sdl/Abstract/Model.js`:
...
 ### GSTREAMER Setup
 
-First, we must determine what gstreamer command works in your environment
+It is easier to determine which gstreamer video sink will work in your environment by testing with a static file. This can be done by downloading [this file](https://support.apple.com/library/APPLE/APPLECARE_ALLGEOS/HT1425/sample_iPod.m4v.zip) and trying the following command.
 
-Start by finding a raw h.264 file ([Example](https://support.apple.com/library/APPLE/APPLECARE_ALLGEOS/HT1425/sample_iPod.m4v.zip)) and determine which of the these gstreamer commands successfully plays the example video
+Common values for sink:
+- ximagesink (x visual environment sink)
+- xvimagesink (xv visual environment sink)
+- cacasink (ascii art sink)
```
[the remark this answers is not in the document]
Comment (iCollin) [reply docs/Feature Documentation/Audio and Video Streaming Guide/index.md:29]: i use it all the time for testing

File: docs/Feature Documentation/Audio and Video Streaming Guide/index.md [hunk 4]
```diff
@@ -31,54 +21,46 @@ Comment out the following lines in `app/model/sdl/Abstract/Model.js`:
...
-or
-```
-gst-launch-1.0 filesrc location=/path/to/h264/file ! decodebin ! videoconvert ! ximagesink sync=false
+gst-launch-1.0 filesrc location=/path/to/h264/file ! decodebin ! videoconvert ! <sink> sync=false
 ```
 
-If you're using tcp, you can connect the stream directly with your phone's IP address using
+If you're streaming video over TCP, you can point gstreamer directly to your phone's stream using
 ```
 gst-launch-1.0 tcpclientsrc host=<Device IP Address> port=3000 ! decodebin ! videoconvert ! ximagesink sync=false
```
[the remark this answers is not in the document]
Comment (iCollin) [reply docs/Feature Documentation/Audio and Video Streaming Guide/index.md:37]: ?

File: docs/Feature Documentation/Audio and Video Streaming Guide/index.md [hunk 5]
```diff
@@ -1,28 +1,18 @@
 ## Initial Configuration
 
-### SDL Core Setup
-
-If you don't already have Core built, you can follow the [Install and Run Guide](../../getting-started/install-and-run/).
```
[the remark this answers is not in the document]
Comment (iCollin) [reply docs/Feature Documentation/Audio and Video Streaming Guide/index.md:5]: This guide was cluttered and a bit hard to follow so I removed it trying to clean up the flow. You are welcome to add it back, I just didn't think it fit in this guide.

File: docs/Feature Documentation/Audio and Video Streaming Guide/index.md (hunk 1 above)
Comment (iCollin) [reply docs/Feature Documentation/Audio and Video Streaming Guide/index.md:43]: Thanks! Updated in https://github.com/smartdevicelink/sdl_core_guides/pull/82/commits/c7b99869d92a546102ef6fd25d556729657e72d6

File: docs/Feature Documentation/Audio and Video Streaming Guide/index.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jacobkeeler) [reply docs/Feature Documentation/Audio and Video Streaming Guide/index.md:5]: Eh, I think it needs to at least be mentioned, since it is a prerequisite for everything in this guide. Perhaps I will add it in as a smaller footnote

File: docs/Feature Documentation/Encryption Setup/index.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+                "keep_context": false,
+                "steal_focus": false,
+                "priority": "NONE",
+                "default_hmi": "NONE",
+                "groups": ["Base-4", "EncryptedAddCommand"],
+                "RequestType": [],
+                "RequestSubType": [],
+                "encryption_required": true
+            }
+...
```
Comment (jacobkeeler) [inline docs/Feature Documentation/Encryption Setup/index.md:124]:
```suggestion
        "functional_groupings": {
            "EncryptedAddCommand": {
                "encryption_required" : true,
                "rpcs":{
                    "AddCommand": {
                        "hmi_levels": ["BACKGROUND",
                        "FULL",
                        "LIMITED"]
                    }
                }
            },
            ...
        },
        ...
        "app_policies": {
            "<PUT_APP_ID_HERE>": {
                "keep_context": false,
                "steal_focus": false,
                "priority": "NONE",
                "default_hmi": "NONE",
                "groups": ["Base-4", "EncryptedAddCommand"],
                "RequestType": [],
                "RequestSubType": [],
                "encryption_required": true
            },
            ...
        }
```

File: docs/Feature Documentation/Encryption Setup/index.md [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+            "encryption_required": true
+        },
+        ...
+    }
+...
+```
+
+#### JSON Example
+
+Below is a possible policy table configuration for requiring a functional group of RPCs to require encryption.
```
Comment (jacobkeeler) [inline docs/Feature Documentation/Encryption Setup/index.md:97]:
```suggestion
Below is a possible policy table configuration requiring an app to use encryption for a specific functional group.
```
?

File: docs/Feature Documentation/Encryption Setup/index.md [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+- Create Public Certificate:
+```bash
+openssl x509 -hash -req -in client.req -signkey client.key -out client.cert -days 10000
+```
+
+## Configure SDL Core
+
+### INI file modifications
+
+- Copy client.key and client.cert into your SDL Core `build/bin` directory. Delete any existing key, crt, or pem files.
```
Comment (jacobkeeler) [inline docs/Feature Documentation/Encryption Setup/index.md:24]:
```suggestion
- Copy client.key and client.cert into your SDL Core `build/bin` directory. Delete any existing key, cert/crt, or pem files.
```

File: docs/Feature Documentation/Encryption Setup/index.md [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                },
+                ...
+            }
+        },
+        ...
+    }
+...
+``` 
+
+- Add `"encryption_required": true` to the app_policies in `sdl_preloaded_pt.json`(or via a [PTU](https://smartdevicelink.com/en/guides/sdl-overview-guides/policies/overview/#policy-table-updates))
```
Comment (jacobkeeler) [inline docs/Feature Documentation/Encryption Setup/index.md:74]:
```suggestion
- Add `"encryption_required": true` to an application in the `app_policies` section
```

File: docs/Feature Documentation/Encryption Setup/index.md [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+- If you are using self signed certificates set `VerifyPeer` to false.
+```ini
+; Verify Mobile app certificate (could be used in both SSLMode Server and Client)
+VerifyPeer  = false
+```
+
+### Policy table modifications
+
+- Add `"encryption_required": true` to a functional group in `sdl_preloaded_pt.json`(or via a [PTU](https://smartdevicelink.com/en/guides/sdl-overview-guides/policies/overview/#policy-table-updates))
```
Comment (jacobkeeler) [inline docs/Feature Documentation/Encryption Setup/index.md:47]:
```suggestion

These modifications can be made in your `sdl_preloaded_pt.json` before launching Core or by updating the policy table while Core is running via a [PTU](https://smartdevicelink.com/en/guides/sdl-overview-guides/policies/overview/#policy-table-updates)

- Add `"encryption_required": true` to a functional group in the `functional_groupings` section
```

File: docs/Feature Documentation/RPC Encryption/index.md [hunk 11]
```diff
@@ -0,0 +1,143 @@
+# RPC Encryption
+
+#### Relevant Evolution Proposals
+- [0207: RPC Message Protection](https://github.com/smartdevicelink/sdl_evolution/blob/master/proposals/0207-rpc-message-protection.md)
+
+## Introduction
+
+This guide will cover the basic setup required to enable and utilize RPC encryption within SDL Core. For more information about the feature, please take a look at the [Encryption Overview Guide](https://smartdevicelink.com/en/guides/sdl-overview-guides/encryption-guide).
```
Comment (jacobkeeler) [inline docs/Feature Documentation/RPC Encryption/index.md:8]:
```suggestion
This guide will cover the basic setup required to enable and utilize RPC Encryption within SDL Core. For more information about the feature, please take a look at the [Encryption Overview Guide](https://smartdevicelink.com/en/guides/sdl-overview-guides/encryption-guide).
```
